code/datums/elements/ridable.dm

```diff
@@ -20,6 +20,9 @@
 	if(!ismovable(target))
 		return COMPONENT_INCOMPATIBLE
 
+	if(!isliving(target))
+		target.pixel_z = WALLENING_OFFSET
```

Comment:
thought 1: can we reset the rider's pixel offsets? it's a bit odd that mounting like a jannie card instashifts it up
thought 2: if not, this should likely be modifying base_pixel to ensure animations in future work properly

Reply:
If u mean to shift them up only when mounted, then that’s doable

Reply:
ok so ideally things that are ridable will have their own existing offset, and placing a mob on them will set the mob to match said offset so they sit properly.

Ideally the mob's offset would not matter here (I think?), like getting onto a mount should place you on the mount rather then moving the mount to match your existing shift

Reply:
Ah. With this current implementation, vehicles will permanently have the offset (since this element is only applied to them once on init). They won’t shift when mounted/dismounted

Reply:
oh I see. that makes more sense but it still feels a bit weird, like vehicles should be allowed to offset up however much and be ridden by mobs offset up however much. It feels kinda hacky but I guess I'm asking for a refactor here huh

Reply:
am good to merge this if you're happy, tho also you should add the offset to base pixel to ensure animations work. are you ok if you skew your pr a bit?

Reply:
Should be good to go